apps/emqx_utils/src/emqx_utils_ets.erl

```diff
@@ -23,13 +23,18 @@
 
 -export([
     lookup_value/2,
-    lookup_value/3
+    lookup_value/3,
+    lookup_value/4
 ]).
 
 -export([keyfoldl/3]).
 
 -export([delete/1]).
 
+-export_type([pat/0]).
+
+-type pat() :: '_' | '$1' | '$2'.
```

Comment:
we also may make it take an argument to clarify intentions
```
-type matchable(T) :: T | '_' | '$1' | '$2'.
```

Reply:
I actually tried that first, but it looked more heavyweight, so I backed down.